src/main/java/xyz/nucleoid/extras/integrations/relay/ChatRelayIntegration.java

```diff
@@ -45,15 +46,19 @@ public static void bind(NucleoidIntegrations integrations, IntegrationsConfig co
             ServerTickEvents.END_SERVER_TICK.register(integration::tick);
 
             if (FabricLoader.getInstance().isModLoaded("styledchat")) {
-                StyledChatEvents.MESSAGE_CONTENT_SEND.register((message, sender, filtered) -> {
-                    if (!filtered) {
-                        integration.onSendChatMessage(sender, message.getString());
+                StyledChatEvents.MESSAGE_CONTENT.register((message, context) -> {
+                    if (context.hasPlayer()) {
+                        var text = message.toText(ParserContext.of(PlaceholderContext.KEY, context), true);
+                        var player = context.player();
+                        assert player != null;
+                        var sender = new MessageSender(player.getUuid(), player.getPlayerListName());
```

Comment:
```suggestion
                        var sender = player.getMessageSender();
```
Should be equivalent (assuming this player is a `PlayerEntity`?)